### DISCUSSION TITLE
username is required when create user in Go

### DISCUSSION THREAD

**Discussion by gaetan1903** [pocketbase/pocketbase#4227] [Q&A]
![image](https://github.com/pocketbase/pocketbase/assets/43904633/9a01e2b9-4244-4ded-91f0-d81e295654ca)

```go
	collection, err := app.Dao().FindCollectionByNameOrId("users")
			if err != nil {
				log.Println("find collection error", err)
				return apis.NewBadRequestError("", nil)
			}

			record0 := models.NewRecord(collection)
			record0.Set("tel", numero)
			record0.SetPassword(security.RandomStringWithAlphabet(16, "a0123456789BcgDehEDJK@LMNFGHIjklmnoPQpqrRSTuvVWXwxyzY"))
			record0.Set("verified", false)

			if err := app.Dao().SaveRecord(record0); err != nil {
				log.Println("save error", err)
				return apis.NewBadRequestError("", nil)
			}
```

**Comment by ganigeorgiev** [comment]
You can set a random generated username or use the [`forms.RecordUpsert`](https://pocketbase.io/docs/go-records/#create-new-record-with-data-validations) and it should create it for you (it will also check for data validation errors).

The system fields (email, username, etc.) will be refactored in the future and devs will have more control over them.

**Reply by ganigeorgiev** [reply]
_I'll update the docs example sometime later to show generating the username._

**Comment by gaetan1903** [comment]
okey, i generated a random username, 

Thank you
